qcengine/programs/psi4.py

```diff
@@ -34,6 +36,28 @@ class Config(ProgramHarness.Config):
 
     @staticmethod
     def found(raise_error: bool = False) -> bool:
+        psithon = which("psi4", return_bool=True)
+        psiapi = which_import("psi4", return_bool=True)
+
+        if psithon and not psiapi:
+            with popen([which("psi4"), "--pythonpath"]) as exc:
+                exc["proc"].wait(timeout=30)
+            if "pythonpath does not exist" in exc["stderr"]:
+                pass
+            else:
+                sys.path.append(exc["stdout"].split()[-1])
+
+        if psiapi and not psithon:
+            psiimport = str(Path(which_import("psi4")).parent.parent)
+            env = os.environ.copy()
+            env["PYTHONPATH"] = psiimport
+            with popen(["python", "-c", "import psi4; print(psi4.executable[:-5])"], popen_kwargs={"env": env}) as exc:
+                exc["proc"].wait(timeout=30)
+            os.environ["PATH"] += os.pathsep + exc["stdout"].split()[-1]
+
+        if psithon or psiapi:
```

Comment:
How costly is all of this logic? We have been having issues where trying to launch Psi4 concurrently on hundreds of nodes can swamp a NAS.

Reply:
It's a hit. 0.4s for `psi4 --module`, 0.6s for `psi4.executable`. The latter isn't reduced by plain `import` instead of `subprocess` b/c `import psi4` is about 0.6s. Good news is that if you set it up the environment right beforehand (as a conda install would), then `Total time: 0.000456 s` for `qcng.get_program('psi4')`.